src/lib/input/input-container.ts

```diff
@@ -314,6 +354,30 @@ export class MdInputContainer implements AfterContentInit {
   /** Focuses the underlying input. */
   _focusInput() { this._mdInputChild.focus(); }
 
+  /** Whether the input container is in an error state. */
+  _isErrorState(): boolean {
+    const control = this._mdInputChild._ngControl;
+    const isInvalid = control ? control.invalid : false;
+    const isTouched = control ? control.touched : false;
+    const isSubmitted = this._parentForm ? this._parentForm.submitted : false;
+
+    return isInvalid && (isTouched || isSubmitted);
+  }
+
+  /** Determines whether to display hints, errors or no messages at all. */
+  _getDisplayedMessages(): 'error'|'hint'|'none' {
+    if (this._errorChildren.length > 0) {
+      if (this._isErrorState()) {
+        return 'error';
+      } else if (this._mdInputChild._ngControl) {
+        let control = this._mdInputChild._ngControl;
+        return (control.valid && control.touched) ? 'none' : 'hint';
```

Comment:
In that case we shouldn't show any messages at all (neither errors nor hints). It's how M1 handles it.